[lsfd] tests failed

Hello!

@masatake 

sparc64 host:
```
tests$ git describe
v2.37-648-g1b44d0923

tests$ ./run.sh --show-diff lsfd

-------------------- util-linux regression tests --------------------

                    For development purpose only.
                 Don't execute on production system!

       kernel: 5.16.0-rc2

      options: --show-diff \
               --srcdir=/1/mator/util-linux/tests/.. \
               --builddir=/1/mator/util-linux/tests/..

         lsfd: directory                      ... SKIPPED (no root permissions)
         lsfd: directory                      ... OK
         lsfd: pipe, no fork                  ... OK
         lsfd: read-only regular file         ...
diff-{{{
--- /1/mator/util-linux/tests/expected/lsfd/mkfds-ro-regular-file       2021-11-26 16:29:39.397686937 +0300
+++ /1/mator/util-linux/tests/output/lsfd/mkfds-ro-regular-file 2021-11-26 17:57:41.055617042 +0300
@@ -13,4 +13,4 @@
 MNTID[RUN]: 0
 DEV[RUN]: 0
 FINDMNT[RUN]: 0
-DEV[STR]: 0
+DEV[STR]: 1
}}}-diff

 FAILED (lsfd/mkfds-ro-regular-file)
         lsfd: character device with O_RDWR   ... OK
         lsfd: AF_UNIX socket pair created with socketpair(2) ...
diff-{{{
--- /1/mator/util-linux/tests/expected/lsfd/mkfds-socketpair    2021-11-26 16:29:39.397686937 +0300
+++ /1/mator/util-linux/tests/output/lsfd/mkfds-socketpair      2021-11-26 17:57:41.831650266 +0300
@@ -1,3 +1,3 @@
-    3  rw- SOCK sockfs      UNIX
-    4  rw- SOCK sockfs      UNIX
+    3  rw- SOCK sockfs UNIX-STREAM
+    4  rw- SOCK sockfs UNIX-STREAM
 ASSOC,MODE,TYPE,SOURCE,PROTONAME: 0
}}}-diff

 FAILED (lsfd/mkfds-socketpair)
         lsfd: symbolic link itself opened with O_PATH ... OK
         lsfd: directory                      ... SKIPPED (no root permissions)
         lsfd: directory                      ... OK

---------------------------------------------------------------------
  2 tests of 9 FAILED
---------------------------------------------------------------------
```

ppc64 host:
```
tests$ ./run.sh --show-diff lsfd

-------------------- util-linux regression tests --------------------

                    For development purpose only.
                 Don't execute on production system!

       kernel: 5.15.0-1-powerpc64

      options: --show-diff \
               --srcdir=/home/mator/util-linux.git/tests/.. \
               --builddir=/home/mator/util-linux.git/tests/..

         lsfd: directory                      ... SKIPPED (no root permissions)
         lsfd: directory                      ... OK
         lsfd: pipe, no fork                  ... OK
         lsfd: read-only regular file         ... OK
         lsfd: character device with O_RDWR   ...
diff-{{{
--- /home/mator/util-linux.git/tests/expected/lsfd/mkfds-rw-character-device    2021-11-26 18:08:03.405921213 +0300
+++ /home/mator/util-linux.git/tests/output/lsfd/mkfds-rw-character-device      2021-11-26 19:08:30.724078893 +0300
@@ -3,4 +3,4 @@
 MNTID[RUN]: 0
 DEV[RUN]: 0
 FINDMNT[RUN]: 0
-DEV[STR]: 0
+DEV[STR]: 1
}}}-diff

 FAILED (lsfd/mkfds-rw-character-device)
         lsfd: AF_UNIX socket pair created with socketpair(2) ...
diff-{{{
--- /home/mator/util-linux.git/tests/expected/lsfd/mkfds-socketpair     2021-11-26 18:08:03.405921213 +0300
+++ /home/mator/util-linux.git/tests/output/lsfd/mkfds-socketpair       2021-11-26 19:08:30.868079286 +0300
@@ -1,3 +1,3 @@
-    3  rw- SOCK sockfs      UNIX
-    4  rw- SOCK sockfs      UNIX
+    3  rw- SOCK sockfs UNIX-STREAM
+    4  rw- SOCK sockfs UNIX-STREAM
 ASSOC,MODE,TYPE,SOURCE,PROTONAME: 0
}}}-diff

 FAILED (lsfd/mkfds-socketpair)
         lsfd: symbolic link itself opened with O_PATH ... OK
         lsfd: directory                      ... SKIPPED (no root permissions)
         lsfd: directory                      ... OK

---------------------------------------------------------------------
  2 tests of 9 FAILED
---------------------------------------------------------------------
```

@masatake 

meanwhile, can we have the following patch applied as well, or i should make it as PR?

```
[mator@centos7vm1 util-linux]$ git diff
diff --git a/configure.ac b/configure.ac
index aa9ad51..df19aa0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -308,6 +308,7 @@ AC_CHECK_HEADERS([ \
        linux/falloc.h \
        linux/fd.h \
        linux/fiemap.h \
+       linux/kcmp.h \
        linux/net_namespace.h \
        linux/nsfs.h \
        linux/raw.h \
@@ -455,6 +456,7 @@ dnl
 have_linux_blkzoned_h=$ac_cv_header_linux_blkzoned_h
 have_linux_btrfs_h=$ac_cv_header_linux_btrfs_h
 have_linux_capability_h=$ac_cv_header_linux_capability_h
+have_linux_kcmp_h=$ac_cv_header_linux_kcmp_h
 have_linux_raw_h=$ac_cv_header_linux_raw_h
 have_linux_securebits_h=$ac_cv_header_linux_securebits_h
 have_linux_version_h=$ac_cv_header_linux_version_h
@@ -1660,6 +1662,7 @@ AM_CONDITIONAL([BUILD_LSCPU], [test "x$build_lscpu" = xyes])
 UL_BUILD_INIT([lsfd], [check])
 UL_REQUIRES_LINUX([lsfd])
 UL_REQUIRES_BUILD([lsfd], [libsmartcols])
+UL_REQUIRES_HAVE([lsfd], [linux_kcmp_h], [linux/kcmp.h header file])
 AM_CONDITIONAL([BUILD_LSFD], [test "x$build_lsfd" = xyes])

 AC_ARG_ENABLE([lslogins],
```

since it is impossible to compile util-linux with `lsfd` on centos-7, when `linux/kcmp.h` is missing... 

```
util-linux$ make lsfd
  CC       misc-utils/lsfd-lsfd.o
misc-utils/lsfd.c:36:24: fatal error: linux/kcmp.h: No such file or directory
 #include <linux/kcmp.h>
                        ^
compilation terminated.
make: *** [misc-utils/lsfd-lsfd.o] Error 1
```

Thanks!

--- a/tests/ts/lsfd/mkfds-socketpair
+++ b/tests/ts/lsfd/mkfds-socketpair
@@ -32,7 +32,7 @@ FD1=4
 EXPR=
 
 {
-    coproc MKFDS { "$TS_HELPER_MKFDS" socketpair $FD0 $FD1 socktype=STREAM; }
+    coproc MKFDS { "$TS_HELPER_MKFDS" socketpair $FD0 $FD1 socktype=DGRAM; }
     if read -u ${MKFDS[0]} PID; then
 	EXPR='(PID == '"${PID}"') and ((FD == '"$FD0"') or (FD == '"$FD1"'))'
 	${TS_CMD_LSFD} -n -o ASSOC,MODE,TYPE,SOURCE,PROTONAME -Q "${EXPR}"

